### DISCUSSION TITLE
Link to earnapp

### DISCUSSION THREAD

**Discussion by af0rte** [engageub/InternetIncome#111] [General]
Hello,
 i  configured the properties file to use proxies, earnapp and logs. 
In the proxies.txt i set two proxies and the work correctly.
![image](https://github.com/engageub/InternetIncome/assets/36008040/4c2d84b5-8f30-423a-a5e8-3cf80ad84819)

The containers  are run correctly 
![image](https://github.com/engageub/InternetIncome/assets/36008040/84307e1d-632f-4dda-b838-b44ab173e78a)

earnapp.txt i have two links
![image](https://github.com/engageub/InternetIncome/assets/36008040/741a0ed3-d237-4f41-9a86-0931e371882d)

when i put the url on the earnapp website i have this error:
![image](https://github.com/engageub/InternetIncome/assets/36008040/07be662a-8d08-41c5-8a84-bb68ba39ee7b)

How can I overcome this error? How can I link the container to earnapp website?

Thanks!

**Comment by engageub** [comment]
Hi,
Could you please let me know if you enabled USE_SOCKS5_DNS or USE_DNS_OVER_HTTPS?
If you wish to debug what's wrong, you will have to enable logs in properties.conf file. 
After enabling logs, you will have to run the following command using the container names or ids. 
Please replace <CONTAINER_NAME OR CONTAINER_ID HERE> with earnapp container name or container id in the following command. 
```
sudo docker container logs <CONTAINER_NAME OR CONTAINER_ID HERE>

```

Paste the output here to understand what's wrong. 

Thank you

**Reply by af0rte** [reply]
thanks for answering me.

i enable USE_SOCKS5_DNS = true on the file properties.conf

i use the comand you suggested  to view logs and i got this following error: 
**sudo docker container logs f3c59360a0e9**

error;
![image](https://github.com/engageub/InternetIncome/assets/36008040/56a8b0a1-4c30-4d47-a97e-7cd0ae97430f)

Thanks.

**Reply by engageub** [reply]
Thank you for the screenshot. The errors indicate that DNS has been failed. Looks like your socks5 proxy does not support DNS.
Please set USE_SOCKS5_DNS to false and try it. 

Thank you

**Reply by af0rte** [reply]
I have put USE_SOCKS5_DNS = false and after reboot the linux and do updateProxies.sh.
I have this error:

![image](https://github.com/engageub/InternetIncome/assets/36008040/db958725-8bb8-4545-a865-a936308c671d)

**Reply by engageub** [reply]
Hello,
Could you please use --delete and then --start instead of update proxies as the old configuration for proxies is still being used.

Thank you

**Reply by engageub** [reply]
Hello,
Its not required to set DNS configuration by default. 
Is your proxy working fine?
Please use --delete when you change something in properties.conf  and then use --start to start the containers.
Please make sure your proxy is working fine. 
To check if your proxy is working fine, please use the following command.
```
curl --proxy https://teste1:test1@139.190.231.xxx.36519/ https://google.com

```

To check if DNS is not blocked on your host, please run the following command

```
nslookup google.com 8.8.8.8

```

Thank you

**Comment by af0rte** [comment]
I have resolve this issue, thank you for helping me!

**Reply by engageub** [reply]
Good to know its working. 
I have seen screenshots from the email which may have been deleted here. If tun2socks containers show restarting, it usually indicates that you have not set the proxy in correct format in proxies.txt file.
If stale containers are running, it indicates you may have not used --delete command or stopped the script while running.
Please read [FAQ](https://github.com/engageub/InternetIncome?tab=readme-ov-file#-how-to-clean-up-all-docker-containers-on-the-host) on how to delete all containers on system. 

Thank you
